Snake/DatabaseLayer/Database.cs

```diff
@@ -1,36 +1,74 @@
-﻿using System.IO;
+﻿using System.Data;
+using System.Data.SqlClient;
+using System.Collections.Generic;
 
 namespace DatabaseLayer
 {
     public class Database
     {
+        private const int MaxNoOfScores = 5;
+        private SqlConnection sqlConnection;
+        private SqlCommand cmd;
+        private SqlDataReader sqlDataReader;
+        private string mazeLevel;
+
+        public Database(string mazeLevel)
+        {
+            sqlConnection = new SqlConnection("Server= localhost; Database= SnakeHighScores; Integrated Security=True;");
+            cmd = new SqlCommand();
+            cmd.CommandType = CommandType.Text;
+            cmd.Connection = sqlConnection;
+            this.mazeLevel = mazeLevel;
+        }
+
         const string FILENAME = "highscores.txt";
 
-        public int getHighSCore()
+        public List<HighScore> getHighSCore()
         {
-            int highScore;
+            cmd.CommandText = "SELECT * FROM HighScores where GameType = '" + mazeLevel + "'";
+            List<HighScore> highScoreList = new List<HighScore>(5);
+
+            sqlConnection.Open();
+            sqlDataReader = cmd.ExecuteReader();
 
-            if (File.Exists(FILENAME))
+            if (sqlDataReader.HasRows)
             {
-                using (BinaryReader reader = new BinaryReader(File.Open(FILENAME, FileMode.Open)))
+                while (sqlDataReader.Read())
                 {
-                    highScore = reader.ReadInt32();
+                    HighScore tempHSObject = new HighScore();
+                    tempHSObject.score = sqlDataReader.GetInt32(2);
+                    tempHSObject.name = sqlDataReader.GetString(3);
+                    highScoreList.Add(tempHSObject);
                 }
             }
-            else
-            {
-                return 0;
-            }
 
-            return highScore;
+            sqlConnection.Close();
+
+            return highScoreList;
         }
 
-        public bool setHighScore(int highScore)
+        public bool setHighScore(List<HighScore> highScoreList)
         {
-            using (BinaryWriter writer = new BinaryWriter(File.Open(FILENAME, FileMode.Create)))
+            cmd.CommandText = "DELETE FROM HighScores where GameType='" + mazeLevel + "'";
+            sqlConnection.Open();

```

Comment:
Would like to see a using statement here so that the connection is disposed of correctly.